src/action_node.cpp

```diff
@@ -182,16 +183,23 @@ NodeStatus BT::AsyncActionNode::executeTick()
             {
                 std::cerr << "\nUncaught exception from the method tick(): ["
                           << registrationName() << "/" << name() << "]\n" << std::endl;
+                // Set the exception pointer and the status atomically.
+                lock_type l(m_);
                 exptr_ = std::current_exception();
-                thread_handle_.wait();
+                setStatus(BT::NodeStatus::FAILURE);
```

Comment:
```suggestion
                setStatus(BT::NodeStatus::IDLE);
```

Reply:
Will do - but I need likely to adjust the tests as well